### PR TITLE
internal/docs/design.md: document principle of consistency

### PR DESCRIPTION
**Pull request by zombiezen**
Emphasized its application to concrete and user-facing types.

### CODE REVIEW

File: internal/docs/design.md [hunk 1]
```diff
@@ -1,7 +1,14 @@
 # Design Decisions
 
 This document outlines important design decisions made for this repository and
-attempts to provide succinct rationales.
+attempts to provide succinct rationales. Recording these decisions helps
+maintain consistency across packages, especially as an open source project
+where contributors can join at any point during development.
+
+A broad design goal for Go Cloud is for the API style to be cohesive and
+consistent: consistency aides users in building a mental model of how to use
```
Comment (jba) [inline internal/docs/design.md]: s/aides/aids/

File: internal/docs/design.md [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+the APIs. As such, the design of individual packages must also consider their
```
Comment (jba) [inline internal/docs/design.md]: You say why consistency is good but not cohesiveness. Maybe you mean the same thing by these two words, in which case pick one.

File: internal/docs/design.md (hunk 1 above)
Comment (zombiezen) [reply internal/docs/design.md]: Done.

File: internal/docs/design.md (hunk 2 above)
Comment (zombiezen) [reply internal/docs/design.md]: Picked "consistency".

**Pull request merged**